### PR TITLE
Version 2.0

### PR DESCRIPTION
**Pull request by willpower232**
- [x] handle config internally to provide more options

- [x] include model and migration for basic email and webhook tables

- [x] include optional ability to record emails directly to database within the package and provide ~new row IDs~ email message ID back to laravel instead of full Postal response

- [x] include optional ability to store and acknowledge webhooks within the package

Anything else?

### CODE REVIEW

File: src/PostalNotificationChannel.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+class PostalNotificationChannel extends MailChannel
+{
+    /**
+     * Send the given notification.
+     *
+     * @param  mixed  $notifiable
+     * @param  \Illuminate\Notifications\Notification  $notification
+     * @return void
+     */
+    public function send($notifiable, Notification $notification)
```
Comment (Josh-G) [inline src/PostalNotificationChannel.php]: can we type hint `$notifiable` as Illuminate\Notifications\Notifiable?

File: src/PostalNotificationChannel.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * Get the recipients of the given message.
+     *
+     * @param  mixed  $notifiable
+     * @param  \Illuminate\Notifications\Notification  $notification
+     * @param  \Illuminate\Notifications\Messages\MailMessage  $message
+     * @return mixed
+     */
+    protected function getRecipients($notifiable, $notification, $message)
```
Comment (Josh-G) [inline src/PostalNotificationChannel.php]: type hint these to?

File: src/PostalNotificationChannel.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+     * Build the mail message.
+     *
+     * @param  \Illuminate\Mail\Message  $mailMessage
+     * @param  mixed  $notifiable
+     * @param  \Illuminate\Notifications\Notification  $notification
+     * @param  \Illuminate\Notifications\Messages\MailMessage  $message
+     * @return void
+     */
+    protected function buildMessage($mailMessage, $notifiable, $notification, $message)
```
Comment (Josh-G) [inline src/PostalNotificationChannel.php]: same again

**Comment by willpower232** [comment]
@Josh-G these methods are overwritten from the [MailChannel class](https://github.com/laravel/framework/blob/5.8/src/Illuminate/Notifications/Channels/MailChannel.php) its extending from so we can't change the declarations.

image

I only had to do that to remove the mail-specific checks in those specific functions otherwise I would have left them alone.

File: config/postal.php [hunk 4]
```diff
@@ -29,5 +29,9 @@
     'webhook' => [
         // route to receive webhooks, configure to avoid collisions with the rest of your app
         'route' => env('POSTAL_WEBHOOK_ROUTE', '/postal/webhook'),
+        // attempt to verify the X-Postal-Signature header
+        'verify' => env('POSTAL_WEBHOOK_VERIFY', false),
```
Comment (Josh-G) [inline config/postal.php]: I think this should be secure by default

**Comment by Josh-G** [comment]
lgtm otherwise

**Pull request merged**